library/hls/src/main/java/com/google/android/exoplayer2/source/hls/HlsMediaChunk.java

```diff
@@ -379,6 +382,9 @@ private void feedDataToExtractor(
       } finally {
         nextLoadPosition = (int) (input.getPosition() - dataSpec.position);
       }
+    } catch (UnexpectedDiscontinuityException e) {
+      Log.d(TAG, "UnexpectedDiscontinuityException - recovering by discarding balance of segment", e);
```

Comment:
What does "discarding balance of segment" mean? Does this actually handle recovery in its current form? I'm not sure how it does, if so?

Reply:
The balance of the chunk's samples simply follow the new PTS sequence, so without a fresh timestamp adjuster, they are useless.

With the last checkin the exception is reported as a playback error  (normal error handler in `Loader` wraps it in an `UnexpectedLoaderException` so `DefaultLoadErrorHandlingPolicy.getRetryDelayMsFor()` will see an `UnexpectedLoaderException` then `Loader.DONT_RETRY_FATAL` is returned so the main playback thread is notified)

I'm working on a partial recovery, which would involve skipping at least to the next chunk, like a seek outside of the `SampleQueue`  range (reseting timestampAdjuster).  This works manually if you add code to 'ignore' the error, like:

````java
  @Override
  public LoadErrorAction onLoadError(
      Chunk loadable,
      long elapsedRealtimeMs,
      long loadDurationMs,
      IOException error,
      int errorCount) {
...
    if (error.getCause() instanceof UnexpectedDiscontinuityException) {
      loadErrorAction = Loader.DONT_RETRY;    // Ignore the balance of this chunk
    }

    eventDispatcher.loadError(
        loadable.dataSpec,
...
    return loadErrorAction;
  }
````